contracts/suilend/sources/lending_market.move

```diff
@@ -373,6 +392,51 @@ module suilend::lending_market {
         fulfill_liquidity_request(lending_market, reserve_array_index, liquidity_request, ctx)
     }
 
+    public fun flash_loan_borrow<P, T>(
+        lending_market: &mut LendingMarket<P>,
+        reserve_array_index: u64,
+        clock: &Clock,
+        amount: u64,
+        ctx: &mut TxContext,
+    ): (Coin<T>, FlashLoan) {
+        let liquidity_request = flash_loan_request<P, T>(
+            lending_market,
+            reserve_array_index,
+            clock,
+            amount,
+        );
+
+        fulfill_flash_loan_liquidity_request(lending_market, reserve_array_index, liquidity_request, ctx)
+    }
+
+    public fun flash_loan_repay<P, T>(
+        lending_market: &mut LendingMarket<P>,
+        reserve_array_index: u64,
+        clock: &Clock,
+        flash_loan: FlashLoan,
+        max_repay_coins: &mut Coin<T>,
+        ctx: &mut TxContext,
+    ) {
+        let lending_market_id = object::id_address(lending_market);
+        assert!(lending_market.version == CURRENT_VERSION, EIncorrectVersion);
+
+        let reserve = vector::borrow_mut(&mut lending_market.reserves, reserve_array_index);
+        assert!(reserve::coin_type(reserve) == type_name::get<T>(), EWrongType);
+
+        reserve::compound_interest(reserve, clock);
+
+        let FlashLoan{ amount_to_repay } = flash_loan;
+        let repay_coins = coin::split(max_repay_coins, amount_to_repay, ctx);
+        reserve::repay_liquidity<P, T>(reserve, coin::into_balance(repay_coins), decimal::from(amount_to_repay));
```

Comment:
see here that it repays the amount of repay coins which is amount borrowed + fees. I'm not sure what the correct way to handle fees should be.